functional-samples/tutorial.tabs-manager/README.md

```diff
@@ -0,0 +1,209 @@
+
+Narrow [host permissions](https://developer.chrome.com/docs/extensions/develop/concepts/match-patterns) allow us to protect user privacy by granting elevated permission to <b>specific sites.</b> This will grant access to the `title`, `URL` properties, as well as additional capabilities. Add the highlighted code to the `manifest.json` file:
+
+```json
+{
+  "host_permissions": [
+    "https://developer.chrome.com/*"
+  ]
+}
+```
+
+#### 💡 Query the tabs:
+
+You can retrieve the tabs from specific URLs using the `tabs.query()` method. Create a `popup.js` file and add the following code:
+
+```javascript
+const tabs = await chrome.tabs.query({
+  url: [
+    "https://developer.chrome.com/docs/webstore/*",
+    "https://developer.chrome.com/docs/extensions/*",
+  ]
+});
+```
+
+#### 💡 Focus on a tab:
+
+First, the extension will sort tab names (the titles of the contained HTML pages) alphabetically. Then, when a list item is clicked, it will focus on that tab using `tabs.update()` and bring the window to the front using `windows.update()`. Add the following code to the `popup.js` file:
+
+```javascript
+const collator = new Intl.Collator();
+tabs.sort((a, b) => collator.compare(a.title, b.title));
+
+const template = document.getElementById("li_template");
+const elements = new Set();
+for (const tab of tabs) {
+  const element = template.content.firstElementChild.cloneNode(true);
+
+  const title = tab.title.split("-")[0].trim();
+  const pathname = new URL(tab.url).pathname.slice("/docs".length);
+
+  element.querySelector(".title").textContent = title;
+  element.querySelector(".pathname").textContent = pathname;
+  element.querySelector("a").addEventListener("click", async () => {
+    // need to focus window as well as the active tab
+    await chrome.tabs.update(tab.id, { active: true });
+    await chrome.windows.update(tab.windowId, { focused: true });
+  });
+
+  elements.add(element);
+}
+document.querySelector("ul").append(...elements);
+```
+
+#### 💡 Group the tabs:
+
+The [TabGroups API](https://developer.chrome.com/docs/extensions/reference/api/tabGroups) allows the extension to name the group and choose a background color. Add the `"tabGroups"` permission to the manifest by adding the highlighted code:
+
+```json
+{
+  "permissions": [
+    "tabGroups"
+  ]
+}
+```
+
+In `popup.js`, add the following code to create a button that will group all the tabs using `tabs.group()` and move them into the current window:
+
+```javascript
+const button = document.querySelector("button");
+button.addEventListener("click", async () => {
+  const tabIds = tabs.map(({ id }) => id);
+  if (tabIds.length) {
+    const group = await chrome.tabs.group({ tabIds });
+    await chrome.tabGroups.update(group, { title: "DOCS" });
+  }
+});
+```
+
+## Running this extension
+
+1. Clone this repository.
+2. Load this directory in Chrome as an [unpacked extension](https://developer.chrome.com/docs/extensions/mv3/getstarted/development-basics/#load-unpacked).
```

Comment:
It would be great to expand this to talk about what happens when the extension is installed and a very basic set of steps you are expected to do.

Reply:
@abrehan2 Did you have a chance to look at this?

Reply:
Yes, mentor. What else do you recommend? I will give it a relook to see if I missed anything.

Reply:
At the moment the steps are very generic and would apply to any sample in this repository. It would be nice to talk about what a developer should expect to happen when they install the extension.

Reply:
such as pin the extension or reload the extension after loading it?